packages/core/src/helpers.ts

```diff
@@ -173,4 +178,56 @@ export default class Helpers {
     }
     return value;
   }
+
+  /**
+   * Helper function to remove reserved keywords in required property of TD JSON Schema
+   */
+  static validateThingDescription(td: any) {
+    if(td.required !== undefined) {
+        let reservedKeywords: Array<string> = [ 
+            "title", "@context", "instance", "forms", "security", "href", "securityDefinitions"
+        ]
+        if (Array.isArray(td.required)) {
+            let reqProps: Array<string> =td.required;
+            td.required = reqProps.filter(n => !reservedKeywords.includes(n))
+        } else if (typeof td.required === "string") {
+            if(reservedKeywords.indexOf(td.required) !== -1)
+                delete td.required
+        }
+    }
+
+    if(td.definitions !== undefined){
+        for (let prop in td.definitions) {  
+            this.validateThingDescription(td.definitions[prop])
+        }
+    }
+
+    return td
+  }
+
+  /**
+   * Helper function to validate an ExposedThingInit
+   */
+  public static validateExposedThingInit(data : any) {
+    const td = JSON.parse(JSON.stringify(tdSchema));
+
+    if(data["@type"] !== undefined && data["@type"] == "tm:ThingModel") {
+      return {
+        valid: false,
+        errors: "ThingModel declaration is not supported"
+      };
+    }
+
+    let exposeThingInitSchema = Helpers.validateThingDescription(td);
```

Comment:
One thing that I don't still like, is the fact that each time we perform the `createExposeThingInitSchema` algorithm (i.e. validateThingDescription in your code). can we move :
 ```ts
let exposeThingInitSchema = Helpers.validateThingDescription(td);
const validate = ajv.compile(exposeThingInitSchema);
```
to a dedicated const outside of the body of `validateExposedThingInit`?

Reply:
Please see new changes. Honestly I can't come with a cleaner solution. it's probably due to my lackings in TS :/
Let me know if you have better suggestions to the problem